gridstatus/utils.py

```diff
@@ -345,3 +345,9 @@ def move_cols_to_front(df, cols_to_move):
     for c in cols_to_move:
         cols.remove(c)
     return df[cols_to_move + cols]
+
+
+def check_latest_value_time(df: pd.DataFrame, value_col: str):
+    """Check the latest value time in the DataFrame"""
+    latest_time = df.loc[df[value_col].last_valid_index(), "Time"]
```

Comment:
Can this be simplified to `df['Time'].max()`?

Reply:
I don't think so (that was my initial approach), but typically there is a timestamp for every time in the day because it also has the day ahead forecast for the value. The file updates the value column incrementally for what we care about. So we need the time at which we have the latest value. `max()` would return the 23:55 timestamp each time. 

Example 
![CleanShot 2024-09-30 at 15 09 35@2x](https://github.com/user-attachments/assets/09c3f7f2-5d59-4858-90fb-ff60d3c3ab54)

I haven't checked this comprehensively for all methods that use `_get_historical()` though (`fuelsource`, `demand`, `storage`), so I will do that now. `max()` may work for some, but conceivably this implementation handles both cases.

Reply:
Thanks for the image, that's helpful. I couldn't figure out what `value_col` was being passed from the caller because we call this with `utils.check_latest_value_time(df, file)` where evidently `file` is a `str`.

Reply:
Okay, I feel this works now as a general solution for `storage`, `fuelsource`, and `demand`, which are the 3 files that take this data in. Hard to test outside of production workflows given the tendency to not work in the ~mornings and to know for sure, but methods are working.

![CleanShot 2024-09-30 at 18 23 01@2x](https://github.com/user-attachments/assets/06bf69ed-38a7-4dcb-a4c7-3116d4c25fde)

@kmax12

Reply:
@WillKoehrsen yeah also took me a bit to realize `file` is meaning the file from caiso sans the .csv, in the request URL. 

Also I was assuming that the file and the value_col were going to be the same, but for instance the `demand` file has the column `Current demand`, so I wrote a small mapping dictionary in that util method to get the correct column name depending on the file. That feels incredibly brittle, but at least gets us off performing the morning data surgery.

Reply:
I see this was addressed by adding a `column` parameter to the function which makes sense.